src/Stratis.SmartContracts.Executor.Reflection/Serialization/MethodParameterByteSerializer.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Nethereum.RLP;
+
+namespace Stratis.SmartContracts.Executor.Reflection.Serialization
+{
+    /// <summary>
+    /// Serializes method parameters using RLP-encoded byte arrays.
+    /// </summary>
+    public class MethodParameterByteSerializer : IMethodParameterSerializer
+    {
+        private readonly IContractPrimitiveSerializer primitiveSerializer;
```

Comment:
Because this uses the primitive serializer, the value will be determined by our serializer behaviour, which I believe will be returning the default value in case of null. Could be misleading? Lets discuss.